File: matrix_add.cpp
#include <iostream>
using namespace std;

int** allocateMatrix(int rows, int cols) {
    int** matrix = new int*[rows];
    for (int i = 0; i < rows; i++) {
        matrix[i] = new int[cols];
        for (int j = 0; j < cols; j++) {
            cout << "행렬의 " << i + 1 << "행 " << j + 1 << "열 원소 입력: ";
            cin >> matrix[i][j];
        }
    }
    return matrix;
}
int** addMatrices(int** matrix1, int** matrix2, int rows, int cols) {
    int** result = new int*[rows];
    for (int i = 0; i < rows; i++) {
        result[i] = new int[cols];
        for (int j = 0; j < cols; j++) {
            result[i][j] = matrix1[i][j] + matrix2[i][j];  // 오타 수정: result[i][j]) -> result[i][j]
        }
    }
    return result;
}
void printMatrix(int** matrix, int rows, int cols) {
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            cout << matrix[i][j] << " ";
        }
        cout << endl;
    }
}
void freeMatrix(int** matrix, int rows) {
    for (int i = 0; i < rows; i++) {
        delete[] matrix[i];
    }
    delete[] matrix;
}
int main() {
    int rows, cols;
    cout << "행렬의 행 수를 입력하세요: ";
    cin >> rows;
    cout << "행렬의 열 수를 입력하세요: ";
    cin >> cols;

    int** matrix1 = allocateMatrix(rows, cols);

    int** matrix2 = allocateMatrix(rows, cols);

    int** resultMatrix = addMatrices(matrix1, matrix2, rows, cols);

    cout << "덧셈 결과:" << endl;
    printMatrix(resultMatrix, rows, cols);

    freeMatrix(matrix1, rows);
    freeMatrix(matrix2, rows);
    freeMatrix(resultMatrix, rows);

    return 0;
}

File: matrix_add.rs
use std::io::{self, Write};

// 행렬의 크기를 입력받아서 행렬을 저장하는 함수
fn input_matrix(rows: usize, cols: usize) -> Vec<Vec<i32>> {
    let mut matrix = vec![vec![0; cols]; rows];
    for i in 0..rows {
        for j in 0..cols {
            let mut input = String::new();
            print!("행렬의 {}행 {}열 원소 입력: ", i + 1, j + 1);
            io::stdout().flush().unwrap();
            io::stdin().read_line(&mut input).unwrap();
            let element: i32 = input.trim().parse().unwrap();
            matrix[i][j] = element;
        }
    }
    matrix
}

// 두 행렬을 더하는 함수
fn add_matrices(matrix1: &Vec<Vec<i32>>, matrix2: &Vec<Vec<i32>>) -> Vec<Vec<i32>> {
    let rows = matrix1.len();
    let cols = matrix1[0].len();
    let mut result = vec![vec![0; cols]; rows];
    for i in 0..rows {
        for j in 0..cols {
            result[i][j] = matrix1[i][j] + matrix2[i][j];
        }
    }
    result
}

// 행렬을 출력하는 함수
fn print_matrix(matrix: &Vec<Vec<i32>>) {
    let rows = matrix.len();
    let cols = matrix[0].len();
    for i in 0..rows {
        for j in 0..cols {
            print!("{} ", matrix[i][j]);
        }
        println!();
    }
}

fn main() {
    // 행렬의 크기 입력 받기
    let mut input = String::new();
    print!("행렬의 행 수를 입력하세요: ");
    io::stdout().flush().unwrap();
    io::stdin().read_line(&mut input).unwrap();
    let rows: usize = input.trim().parse().unwrap();

    input.clear();
    print!("행렬의 열 수를 입력하세요: ");
    io::stdout().flush().unwrap();
    io::stdin().read_line(&mut input).unwrap();
    let cols: usize = input.trim().parse().unwrap();

    // 행렬 입력
    let matrix1 = input_matrix(rows, cols);
    let matrix2 = input_matrix(rows, cols);

    // 두 행렬 더하기
    let result_matrix = add_matrices(&matrix1, &matrix2);

    // 결과 출력
    println!("덧셈 결과:");
    print_matrix(&result_matrix);
}
